File: JavaScript/390. Elimination Game.js
/**
 * @param {number} n
 * @return {number}
 */
var lastRemaining = function(n) {
    var l = true;
    var res = 1;
    var step = 1;
    var rem = n;
    while(rem > 1){
        if(l || rem % 2 == 1){
            res += step;
        }
        step *= 2;
        rem =  Math.floor(rem / 2);
        l = !l;
    }
    return res;
};

File: C++/390. Elimination Game.cpp
class Solution {
public:
    int lastRemaining(int n) {
        bool l = true;
        int res = 1;
        int step = 1;
        int rem = n;
        while(rem > 1){
            if(l || rem % 2 == 1){
                res += step;
            }
            step *= 2;
            rem /=  2;
            l = !l;
        }
        return res;
    }
};
